src/sentry/models/groupassignee.py

```diff
@@ -85,11 +96,19 @@ class GroupAssignee(Model):
 
     project = FlexibleForeignKey('sentry.Project', related_name="assignee_set")
     group = FlexibleForeignKey('sentry.Group', related_name="assignee_set", unique=True)
-    user = FlexibleForeignKey(settings.AUTH_USER_MODEL, related_name="sentry_assignee_set")
+    user = FlexibleForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name="sentry_assignee_set",
+        null=True)
+    team = FlexibleForeignKey('sentry.Team', null=True)
     date_added = models.DateTimeField(default=timezone.now)
 
     class Meta:
         app_label = 'sentry'
         db_table = 'sentry_groupasignee'
 
-    __repr__ = sane_repr('group_id', 'user_id')
+    __repr__ = sane_repr('group_id', 'user_id', 'team_id')
+
+    def save(self, *args, **kwargs):
+        assert self.user_id or self.team_id
```

Comment:
I'm thinking about applying a migration as well to put a real constraint on this to enforce it at the database.

Reply:
+1 on making this a defined check constraint; I'd also strongly consider ensuring that only one of the two is set (require of the fields to be set and only allow one of the fields to be set) both here as well as in the check constraint.

I have no idea how we'd hook these check constraints in to to run correctly during tests… add SQL manually to the migration somehow?

Reply:
```sql
ALTER TABLE sentry_groupasignee
ADD CONSTRAINT require_team_or_user_but_not_both
CHECK (
    NOT (team_id IS NOT NULL AND user_id IS NOT NULL) AND
    NOT (team_id IS NULL AND user_id IS NULL)
);
```

😎

Reply:
:nice:

Reply:
Added here: https://github.com/getsentry/sentry/blob/8b29ec7dc85367a73ced7f10c8116f7f3f1286a1/src/sentry/south_migrations/0387_auto__add_field_groupassignee_team__chg_field_groupassignee_user.py#L33-L38